src/ts/content.ts

```diff
@@ -137,21 +133,24 @@ const replaceIcon = ({
     isDirectory = iconDom.classList.contains('octicon-file-directory');
   }
 
-  const getClassName = () => {
+  const getIconColorMode = (): ColorMode => {
     if (colorsDisabled) {
-      return getClass(filename);
+      return 'mono';
     }
 
     if (siteDarkMode) {
-      return getClassWithDarkColor(filename);
+      return 'dark';
     }
 
-    return getClassWithColor(filename);
+    return 'light';
   };
 
-  const className = getClassName();
+  const className = getIconClass(filename, {
+    colorMode: getIconColorMode(),
+    skipFallback: true,
+  });
 
-  if (String(className).includes('icon-file-text')) {
+  if (!className) {
```

Comment:
I'm not sure what you think about this part. I added an option to my package so it will normally return the `icon-file-text` class if there is no match, but if you pass `skipFallback: true`, it will return null. This means that the extension will return the text icon, but only for files that match it like `.txt` files.